### PR TITLE
Add file header template editorconfig

### PR DESCRIPTION
**Pull request by LeafShi1**
## Proposed changes

- Add file_header_template to .editorconfig file for .cs and .vb files
- Change all cs and vb files to use 2-line header

## Risk

- Minimal

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9615)

### CODE REVIEW

**Comment by LeafShi1** [comment]
Wait for the build to complete and the Unit tests to pass.

File: .editorconfig [hunk 1]
```diff
@@ -14,6 +14,11 @@ indent_size = 4
 [*.json]
 indent_size = 2
 
+# C# and VB files
+[src/**/*.{cs,vb}]
+charset = utf-8-bom
```
Comment (Tanya-Solyanik) [inline .editorconfig:19]: `charset` is already defined in the next rule.

File: .editorconfig [hunk 2]
```diff
@@ -14,6 +14,11 @@ indent_size = 4
...
+file_header_template = Licensed to the .NET Foundation under one or more agreements.\nThe .NET Foundation licenses this file to you under the MIT license.\nSee the LICENSE file in the project root for more information.
```
Comment (Tanya-Solyanik) [inline .editorconfig:20]: @JeremyKuhne  - System.Drawing uses a different header, one that does not refer to the `LICENSE` file in the root. Is that desirable, or should it be unified to the winforms header?

Comment (JeremyKuhne) [reply .editorconfig:20]: @Tanya-Solyanik You should reach out to @richlander offline to find the current "right" answer (or the person who knows).

Comment (Tanya-Solyanik) [reply .editorconfig:20]: ok, @LeafShi1 - meanwhile, please exclude System.Drawing from this change.

Comment (Tanya-Solyanik) [reply .editorconfig:20]: @richlander said that the 2-line header is the new one that we should use everywhere. @LeafShi1 - please change the code base to the new header

**Comment by Tanya-Solyanik** [comment]
it's fine to update header for .cpp and .h files

**Comment by LeafShi1** [comment]
> it's fine to update header for .cpp and .h files

ok, I'll change them back.

In addition, there is a problem, when I added the `file_header_template ` for .vb in .editorconfig , the build error "[.dotnet\sdk\8.0.100-preview.6.23330.14\Roslyn\Microsoft.VisualBasic.Core.targets(43,5): error MSB6006: (NETCORE_ENGINEERING_TELEMETRY=Build) "vbc.dll" exited with code 1](https://dev.azure.com/dnceng-public/public/_build/results?buildId=358645&view=logs&j=fdea2036-e6b2-512d-a259-98f88c972586&t=e85b1de4-8dce-50bb-88c3-24524432e33a&l=176)." appeared, so I deleted the .vb, can I only add template for the cs file?

**Review by SimonZhao888 — APPROVED** [review]

**Comment by Tanya-Solyanik** [comment]
>can I only add template for the cs file?

Sure!

**Comment by Tanya-Solyanik** [comment]
Does the VB crash repro locally?

**Comment by LeafShi1** [comment]
> Does the VB crash repro locally?

No

**Review by Epica3055 — DISMISSED** [review]

**Review by Tanya-Solyanik — DISMISSED** [review]

**Comment by Tanya-Solyanik** [comment]
@LeafShi1 - Could you please add a comment to .editorconfig that explains why .vb files are excluded. `Bvc.dll crashed in the CI build when this rule is enabled for VB." This can be done in a new PR.

**Comment by LeafShi1** [comment]
> @LeafShi1 - Could you please add a comment to .editorconfig that explains why .vb files are excluded. `Bvc.dll crashed in the CI build when this rule is enabled for VB." This can be done in a new PR.

OK, I will add that in next PR

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Pull request merged**